### ISSUE TITLE
NewTab Adapter doesn't respect Vivaldi

### ISSUE THREAD

**Issue by hacker-DOM** [gdh1995/vimium-c#530]
Hi there!

So I'm using:

- Vivaldi@5.0.2497.32 
- Vimium C@1.96.1
- NewTab Adapter@1.2.0

These days, Vivaldi has two types of tab openings:
- "New Top Level Tab" - regular new tab
- "New Tab" - opens a tab in the current Tab Stack

It seems to me that with _NewTab Adapter_ the new tab doesn't open as "New Tab", but rather as always as a top-level tab..

My current workaround:

- find a webpage you like, like https://example.com or a social network image.
- set is as the default page
- deactivate NewTab Adapter
- Vimium C now works with Vivaldi!

Still, it would be nice to use _NewTab Adapter_ :)

PS The same seems to be true for most of _Vimium C_'s settings, like "Open URL, history, … in a new tab"

Thanks

**Comment by gdh1995** [comment]
Confirmed. But, it's a private feature of Vivaldi and the browser hasn't provided any special API to do so.

In Vimium C, there's a working around to open URLs in a same tab stack:
```
map t openUrl group opener
```

And most other commands which may open a tab should also support the two options.

As for New Tab Adapter, sorry but it can't now. I changed some code before, but it's not correct enough so I haven't published a new version of it.

I suggest you to use Vimium C's shortcuts (`chrome://extensions/shortcuts`) to create tabs, instead of the default one of your browser. In fact, if you only uses Vimium C's `createTab`, then New Tab Adapte is not necessary at all.

**Comment by hacker-DOM** [comment]
Thanks for your response. I played around with it and here's my current setup :-)

1. Disable browser's (Vivaldi's) keyboard shortcuts for "New Tab" and "New Top Level Tab".
2. Add "CMD-Y" as a keyboard shortcut for "Create new tab" in chrome://extensions/shortcuts. I set it to "Global", it doesn't work when "Vivaldi" is selected...
3. Add the following settings:

For top level tab:
`New Tab URL: pages/blank.html`

Keymappings:
```
# activates vomnibar on new top level tab (cmd-y)
shortcut createTab $then="Vomnibar.activate"
# cmd-t creates a new stack level tab. 'hfj...' is vimium c's id
map <m-t> openUrl group opener url="chrome-extension://hfjbmagddngcpeloejdejnfgbamkjaeg/pages/blank.html" $then="Vomnibar.activate" 
```

I only have two remaining issues for a perfect workflow :)

1. the `$then="Vomnibar.activate"` part doesn't seem to work for `<m-t>`.
2. When the Vomnibar is quit (Esc) on `pages/blank.html`, there is a flash of white space. I am not getting this on any other website. Fwiw I am using `#enable-force-dark: Enabled` in chrome://settings.

**Comment by gdh1995** [comment]
1.
  - `shortcut xxxx` is only a inner feature of Vimium C; while since you bind `Cmd+Y` to it, then `Meta+T` will only trigger Vivaldi's built-in feature of opening a new tab, so it won't make Vimium C executes `createTab` and `Vomnibar.activate`.
  - when I test `Cmd+Y`, Vomnibar also fails. So it may be a compatibility issue of Vimium C. So does Vomnibar on Firefox. Sad.
  - As far as I now imagine, it's not related with the `Focus Page Content on New Tab` option of Vivaldi
2. it sounds like a bug. maybe I need more tests when I in rest.

**Comment by gdh1995** [comment]
Hello, `$then=Vomnibar.*` and the flash of white area on Vomnibar hiding have been fixed in v1.96.3.
